source/common/network/common_connection_filter_states.cc

```diff
@@ -0,0 +1,17 @@
+#include "source/common/network/common_connection_filter_states.h"
+
+namespace Envoy {
+
+const ExecutionContext*
+GetConnectionExecutionContextReadOnly(const Network::Connection& connection) {
+  const ConnectionExecutionContextFilterState* filter_state =
+      connection.streamInfo().filterState().getDataReadOnly<ConnectionExecutionContextFilterState>(
+          kConnectionExecutionContextFilterStateName);
+  return filter_state == nullptr ? nullptr : filter_state->execution_context();
+}
+
+ExecutionContext* GetConnectionExecutionContextMutable(const Network::Connection& connection) {
+  return const_cast<ExecutionContext*>(GetConnectionExecutionContextReadOnly(connection));
```

Comment:
Is this const_cast "safe"? Is it OK to const cast the output from `getDataReadOnly()`?

Reply:
Yes: The execution_context is a `std::unique_ptr` in `ConnectionExecutionContextFilterState`, so it's always a heap-allocated object and safe to const_cast.

The const_cast is hard to avoid because some callers have a `const Network::Connection` at hand.